src/libsystemd/sd-bus/bus-socket.c

```diff
@@ -590,6 +590,15 @@ static int bus_process_cmsg(sd_bus *bus, struct msghdr *mh, bool allow_fds) {
                 return 0;
         }
 
+        if (n_fds > 0 && bus->fds_might_be_truncated)
+                /* We previously received fds with MSG_CTRUNC set, indicating the fd array was truncated
+                 * (e.g., by an LSM blocking some fds). When the kernel truncates the fd array, it drops
+                 * all fds from the blocked one onwards - we have no way to know how many were lost or
+                 * which indexes are affected. Any fds we receive now would be appended at the wrong
+                 * indexes, corrupting the message's fd table. We must silently drop them to avoid a
+                 * worse mess. The message will still fail later if it tries to access a truncated fd. */
```

Comment:
"The message will fail": this sentence is strange. The message is a passive object.